core/src/main/java/org/jdbi/v3/core/ResultBearing.java

```diff
@@ -156,4 +163,49 @@ default T findOnly() {
                 });
         }
     }
+
+    /**
+     * Reduce the results.  Using a {@code BiFunction<U, ReduceRow, U>}, repeatedly
+     * combine query results until only a single value remains.
+     *
+     * @param identity the {@code U} to combine with the first result
+     * @param accumulator the function to apply repeatedly
+     * @return the final {@code U}
+     */
+    default <U> U reduceRows(U identity, BiFunction<U, ReduceRow, U> accumulator) {
+        return execute((mapper, rs, ctx) -> {
+            try {
+                ReduceRow rr = new ReduceRow(rs, ctx);
+                U result = identity;
+                while (rs.next()) {
+                    result = accumulator.apply(result, rr);
+                }
+                return result;
+            } catch (SQLException e) {
```

Comment:
If `StatementExecutor` threw SQLException, you could catch it once in the execute method above.